api/envoy/api/v2/route/route_components.proto

```diff
@@ -145,10 +145,17 @@ message VirtualHost {
   // <envoy_api_field_config.filter.http.router.v2.Router.suppress_envoy_headers>` flag.
   bool include_request_attempt_count = 14;
 
-  // Indicates the retry policy for all routes in this virtual host. Note that setting a
-  // route level entry will take precedence over this config and it'll be treated
-  // independently (e.g.: values are not inherited).
-  RetryPolicy retry_policy = 16;
+  oneof retry_policy_specifier {
```

Comment:
A oneof upgrade is a breaking compile change so we can't do it. I would just add the new field and have a check in code that both can't be specified. There is a oneof upgrade annotation that can be applied so it gets upgraded in the next major version.